### PR TITLE
Write getting started guide for new contributors.

### PR DESCRIPTION
**Pull request by doriath**
Signed-off-by: Tomasz Zurkowski <zurkowski@google.com>

### CODE REVIEW

File: Documentation/getting-started.md
Comment (orangecms) [inline Documentation/getting-started.md:58]: Just without the question mark, since it's not formulated as a question :)

**Review by orangecms — COMMENTED** [review]
Neat, thanks a ton! \o/

I have only skimmed through, but it looks great at first glance. :)

**Review by rminnich — COMMENTED** [review]
if we went x86 we could completely skip the build step for riscv, and also any reference to azure, could we do that?

**Comment by doriath** [comment]
About going with x86 and dropping azure:

- RISC-V is what runs on azure (CI) right now and is guaranteed to work. I found it really useful to have an example, working setup I could compare to. We would have to add x86 to CI before we update this text.
- Even if we go x86, getting azure setup was the most important part of understanding the project. README files can get out of date, but commands in CI are guaranteed to be up to date. For example, I had no idea that I should start my command with `PAYLOAD_A` to actually get `TESTTESTTEST` on the screen. Also, getting the example stdout from azure logs let me know what I should expect from my setup
- It was also important to me to know which QEMU version I should use. I first started with version from head, but it actually didn't work so I had to pin it exactly to the version from CI.

Because of the above, I would like to keep mentioning `azure-pipelines.yml` file if possible.

File: Documentation/getting-started.md
Comment (alistair23) [inline Documentation/getting-started.md:30]: I don't think you need to indicate a tag here. It would be best to just use QEMU master.

File: Documentation/getting-started.md
Comment (doriath) [reply Documentation/getting-started.md]: Done, thanks!

File: Documentation/getting-started.md
Comment (doriath) [reply Documentation/getting-started.md:30]: When I was starting with oreboot, HEAD master of QEMU was actually not working. For a new contributor, I wanted to increase the chance of it working for the first time (later they can experiment with other versions of QEMU).

Comment (orangecms) [reply Documentation/getting-started.md:30]: Is there anything wrong with just using the distro package? I'd not want to have ti build QEMU myself... and then again, why would CI need to build it? For building from source, I would definitely stick with a tagged and presumably stable release.

File: Documentation/getting-started.md
Comment (orangecms) [inline Documentation/getting-started.md:4]: oreboot should be lowercase

File: Documentation/getting-started.md
Comment (orangecms) [inline Documentation/getting-started.md:6]: **a** Debian like system

File: Documentation/getting-started.md
Comment (orangecms) [inline Documentation/getting-started.md:36]: I would suggest symlinking to `/usr/local/bin/` to avoid possible collisions when people install QEMU through their package manager as well.

File: Documentation/getting-started.md
Comment (orangecms) [inline Documentation/getting-started.md:73]: **Rust** should start with an uppercase R

File: Documentation/getting-started.md
Comment (orangecms) [inline Documentation/getting-started.md:115]: "flashes" might be confusing here; how do you feel about "takes"?

File: Documentation/getting-started.md
Comment (orangecms) [inline Documentation/getting-started.md:137]:
```md
* `make flash` - flash to hardware, specify programmer for flashrom with `FLASHROM_PROGRAMMER`
```

File: Documentation/getting-started.md
Comment (alistair23) [reply Documentation/getting-started.md:30]: Distro packages work great, you just need to use a modern distro with v5.1.0 or newer.

Can we update this to tell users to use their distro QEMU if it's v5.1.0 or newer, otherwise build v5.1.0?

Comment (doriath) [reply Documentation/getting-started.md:30]: 1. This example shows RISC-V - is there prebuilt qemu for RISC-V that can be downloaded (I though this is the main reason we build it).
2. The exact version is important for RISC-V, because when I started ~month HEAD master was not working, and I needed a specific old version to have it run properly

I created #295 to setup qemu-q35 in CI. Once that is done, I think we could update this guide to just use distro package and use q35 instead of riscv. WDYT?

File: Documentation/getting-started.md
Comment (doriath) [reply Documentation/getting-started.md]: Good point, thanks!

File: Documentation/getting-started.md
Comment (doriath) [reply Documentation/getting-started.md]: Replaced all `rust` with `Rust`, thanks!

File: Documentation/getting-started.md
Comment (doriath) [reply Documentation/getting-started.md]: Done.

File: Documentation/getting-started.md
Comment (doriath) [reply Documentation/getting-started.md]: Done.

**Review by doriath — COMMENTED** [review]
Thank you for all the comments!

File: Documentation/getting-started.md
Comment (alistair23) [reply Documentation/getting-started.md:30]: If we add q35 testing then we should test both not just 1. It doesn't make a lot of sense to remove test cases from the CI. RISC-V is stabalising quickly so oreboot should be fine to use 5.1+ for the foreseeable future.

Comment (doriath) [reply Documentation/getting-started.md:30]: Yes, we definitely should still test both and we should not remove RISC-V from CI. What I meant is to remove RISC-V from this guide, and just use q35 (there is no need for the user to mention both if q35 is simpler to start with). And I was suggesting to remove risc-v from this guide because of this comment https://github.com/oreboot/oreboot/pull/289#pullrequestreview-492220337 from Ron.

Comment (orangecms) [reply Documentation/getting-started.md:30]: In Arch, we have RISC-V QEMU packaged. 5.1.0 is the current version. See here:
https://www.archlinux.org/packages/extra/x86_64/qemu-arch-extra/

**Comment by orangecms** [comment]
I'd like to propose getting the core thing in and just omit the Q35, RISC-V etc part for now, which can be moved to a follow-up PR. WDYT?

**Comment by doriath** [comment]
> I'd like to propose getting the core thing in and just omit the Q35, RISC-V etc part for now, which can be moved to a follow-up PR. WDYT?

That is a good idea. Removed part of the "Setup environment" section so that we can merge it, and we can discuss risc-v/q35 in a followup.

**Review by orangecms — APPROVED** [review]
I like it, would just like to wait for another approval before merging. 🙂

**Review by rminnich — APPROVED** [review]
This is great. I would suggest getting someone who has never seen this project to try it out.

**Comment by futaris** [comment]
I've seen this project, but never gotten around to setting up a build process, or use oreboot.

I use x86 coreboot in my dayjob and fiddle with some RISC-V FPGA stuff outside of work.

**Comment by futaris** [comment]
Ran through everything. It all worked on a Ubuntu 18.04.5 LTS x86-64 system.

Might be worth mentioning system / disk space requirements. 2GiB space is probably more than enough.

The only other thing that I would test, would be what other build packages might be needed before the git pull.  Either create a VM with a fresh install of a Debian system, or a docker or something similar. Does the user just need to `apt install git` or install anything else?

I haven't tried anything with Qemu, as those instructions aren't there (yet)? I did build oreboot from 
`/src/mainboard/emulation/qemu-q35` without any issues though.

Feel free to ping me on the Q35 / RISC-V running oreboot in Qemu follow-on patch, for review. Either on github or https://slack.osfw.dev/ #oreboot ...

**Pull request merged**
